### PR TITLE
Fix typo INDLE and remove sum function

### PR DESCRIPTION
**Pull request by SSDaniel**
That's all folks.

by MM

### CODE REVIEW

**Comment by pdsfilho** [comment]
swaped engine struct to bsg struct and added additional unit tests

File: code/bsg.o
Comment (akafael) [inline code/bsg.o:1]: Remove binary file include.

File: code/bsg.c [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+  };
+  if (vehicle.angleBrakePedal == 0){
+    return BSG_STARTER;
+  }
+  if (battery.voltage >= 0) {
+      return BSG_STARTER;
+  };
+  if (battery.current >= 0) {
+      return BSG_STARTER;
+  };
```
Comment (akafael) [inline code/bsg.c:22]: Suggestion:  Group all similar ifs using bitwise or operator instead using multiple ifs.

The following code would generate the same behavior:

```suggestion
  if ( (vehicle.angleBrakePedal == 0) || (battery.voltage >= 0) || (battery.current >= 0) ) {
      return BSG_STARTER;
  };
```

File: code/test_bsg.c [hunk 2]
```diff
@@ -2,8 +2,50 @@
...
+  Engine engine;
+  engine.rpm = 0;
+
+  Battery battery;
+  battery.voltage = 0;
+  battery.current = 0;
+
+  const StateBSG bsgMode = selectBSGMode(vehicle, engine, bsg, battery);
+  ASSERT_EQ(bsgMode, BSG_IDLE);
+}
```
Comment (akafael) [inline code/test_bsg.c:28]: Suggestion: Add line break between test cases to improve readability.

```suggestion
}

```

**Review by akafael — APPROVED** [review]

**Pull request merged**